common/remote/hooks/azure/azurefile.cpp

```diff
@@ -601,112 +596,170 @@ AzureFile::AzureFile(const char *_azureFileName) : fullName(_azureFileName)
-    std::istringstream input(std::string((const char *)data, len));
-    //implement append_block_from_buffer based on upload_block_from_buffer
-    auto ret = client->append_block_from_stream(containerName, blobName, input).get();
-    checkError(ret);
+    auto appendBlobClient = getClient<AppendBlobClient>();
+    try
+    {
+        Azure::Core::IO::MemoryBodyStream content(reinterpret_cast <const uint8_t *>(data), len);
+        appendBlobClient->AppendBlock(content);
+    }
+    catch (const Azure::Core::RequestFailedException& e)
+    {
+        IException * error = makeStringExceptionV(1234, "Azure append blob failed: %s (%d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+        throw error;
+    }
 }
 
 void AzureFile::createBlockBlob()
 {
-    auto client = getClient();
-    auto http = client->client()->get_handle();
-
-    auto request = std::make_shared<azure::storage_lite::create_block_blob_request>(containerName, blobName);
-
-    auto ret = async_executor<void>::submit(client->account(), request, http, client->context()).get();
-    checkError(ret);
+    auto blockBlobClient = getClient<BlockBlobClient>();
+    try
+    {
+        Azure::Core::IO::MemoryBodyStream empty(nullptr, 0);
+        Azure::Response<Models::UploadBlockBlobResult> result = blockBlobClient->Upload(empty); // need to do this to create an empty blob
+        setProperties(0, result.Value.LastModified, result.Value.LastModified);
+    }
+    catch (const Azure::Core::RequestFailedException& e)
+    {
+        IException * error = makeStringExceptionV(1234, "Azure create block blob failed: %s (%d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+        throw error;
+    }
 }
 
 
 void AzureFile::appendToBlockBlob(size32_t len, const void * data)
 {
-    auto client = getClient();
-    std::string blockid;
-    auto ret = client->upload_block_from_buffer(containerName, blobName, blockid, (const char *)data, len).get();
-    checkError(ret);
+    auto appendBlobClient = getClient<AppendBlobClient>();
+    try
+    {
+        Azure::Core::IO::MemoryBodyStream content(reinterpret_cast <const uint8_t *>(data), len);
+        appendBlobClient->AppendBlock(content);
+    }
+    catch (const Azure::Core::RequestFailedException& e)
+    {
+        IException * error = makeStringExceptionV(1234, "Azure append block blob failed: %s (%d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+        throw error;
+    }
 }
 
 void AzureFile::commitBlockBlob()
 {
-    auto client = getClient();
-    std::vector<azure::storage_lite::put_block_list_request_base::block_item> blocks;
-    auto ret = client->put_block_list(containerName, blobName, blocks, {}).get();
-    checkError(ret);
-}
-
-std::shared_ptr<azure::storage_lite::blob_client> AzureFile::getClient()
-{
-    return ::getClient(accountName, accountKey);
+    // TODO:
+    // commit block blob requires blocks to be staged first - to stage use BlockBlobClient::StageBlock()
```

Comment:
Should throw UNIMPLEMENTED

Reply:
I'm removing this function as it is unnecessary at the moment.